hazelcast/src/main/java/com/hazelcast/cache/impl/AbstractCacheService.java

```diff
@@ -103,6 +106,7 @@ public final void init(NodeEngine nodeEngine, Properties properties) {
         }
         this.cacheEventHandler = new CacheEventHandler(nodeEngine);
         this.cacheSplitBrainHandler = new CacheSplitBrainHandler(nodeEngine, configs, segments);
+        this.cacheWanEventPublisher = new CacheWanEventPublisherImpl();
```

Comment:
This is just minor comment. OSS `ICacheService` (`CacheService`) doesn't support WAN. So I think, no need to create `CacheWanEventPublisher` for all `ICacheService` implementations. You may define a protected `createCacheWanEventPublisher()` method and only `EnterpriseCacheService` returns a `CacheWanEventPublisher` instance.